src/Microsoft.AspNetCore.Mvc.Core/ControllerBase.cs

```diff
@@ -1038,6 +1038,96 @@ public virtual ForbidResult Forbid(IList<string> authenticationSchemes, Authenti
+        /// <param name="properties"><see cref="AuthenticationProperties"/> used to perform the sign-in operation.</param>
+        /// <returns>The created <see cref="SignInResult"/> for the response.</returns>
+        [NonAction]
+        public virtual SignInResult SignIn(
+            IList<string> authenticationSchemes,
+            ClaimsPrincipal principal,
+            AuthenticationProperties properties)
+            => new SignInResult(authenticationSchemes, principal, properties);
+
+        /// <summary>
+        /// Creates a <see cref="SignOutResult"/> with the specified authentication scheme.
+        /// </summary>
+        /// <param name="authenticationScheme">The authentication scheme to use for the sign-out operation.</param>
+        /// <returns>The created <see cref="SignOutResult"/> for the response.</returns>
+        [NonAction]
+        public virtual SignOutResult SignOut(string authenticationScheme)
+            => new SignOutResult(authenticationScheme);
+
+        /// <summary>
+        /// Creates a <see cref="SignOutResult"/> with the specified authentication schemes.
+        /// </summary>
+        /// <param name="authenticationSchemes">The authentication schemes to use for the sign-out operation.</param>
+        /// <returns>The created <see cref="SignOutResult"/> for the response.</returns>
+        [NonAction]
+        public virtual SignOutResult SignOut(IList<string> authenticationSchemes)
+            => new SignOutResult(authenticationSchemes);
+
+        /// <summary>
+        /// Creates a <see cref="SignOutResult"/> with the specified specified authentication scheme and
+        /// <paramref name="properties" />.
+        /// </summary>
+        /// <param name="authenticationScheme">The authentication scheme to use for the sign-out operation.</param>
+        /// <param name="properties"><see cref="AuthenticationProperties"/> used to perform the sign-out operation.</param>
+        /// <returns>The created <see cref="SignOutResult"/> for the response.</returns>
+        [NonAction]
+        public virtual SignOutResult SignOut(string authenticationScheme, AuthenticationProperties properties)
+            => new SignOutResult(authenticationScheme, properties);
+
+        /// <summary>
+        /// Creates a <see cref="SignOutResult"/> with the specified specified authentication schemes and
+        /// <paramref name="properties" />.
+        /// </summary>
+        /// <param name="authenticationSchemes">The authentication scheme to use for the sign-out operation.</param>
+        /// <param name="properties"><see cref="AuthenticationProperties"/> used to perform the sign-out operation.</param>
+        /// <returns>The created <see cref="SignOutResult"/> for the response.</returns>
+        [NonAction]
+        public virtual SignOutResult SignOut(IList<string> authenticationSchemes, AuthenticationProperties properties)
```

Comment:
I believe this has a similar small flaw in that potentially you might want to pass different AuthenticationProperties to each signout.  Maybe just limit things to a single sign in/sign out which is the most common case?

Any additional sign in/sign outs can be done outside of the result so there's not really any loss of flexibility without these overloads